inspirehep/modules/workflows/workflows/hep_ingestion.py

```diff
@@ -125,6 +130,11 @@ class mysource(HEPIngestion):
             with_author_keywords=True,
         ),
         filter_core_keywords,
+        guess_categories,
+        IF(is_experimental_paper, [
+            guess_experiments,
+        ]),
+        guess_keywords,
```

Comment:
The arXiv harvesting is currently inheriting from the hep ingestion workflow to reuse some common parts. Now the `before_halt_check` part is indeed becoming longer and parts of it is equal between them. We could split `before_halt_check` up to more parts to just add the arXiv parts before/after.

But looking at it again now I am honestly not sure that is the right direction to share the workflow across the two.

In terms of readability and helping understanding what is going on in the different workflows, you see it's already today quite complex to understand what is happening and one must go back and forth between those two files to see the whole picture. The best way today is actually is to open a shell and import the workflow and echo the workflow attribute.

So, I am wondering now if it would just be better to just be more direct in the workflow definitions and not inherit parts at all. Similar to how literature vs. arxiv workflow is today. Or, that we have one workflow with multiple branches (if arxiv etc.). WDYT?